plugins/linear/server/linear.ts

```diff
@@ -0,0 +1,160 @@
+      body,
+    });
+
+    return AccessTokenResponseSchema.parse(await res.json());
+  }
+
+  static async revokeAccess(accessToken: string) {
+    const headers = {
+      Authorization: `Bearer ${accessToken}`,
+    };
+
+    await fetch(LinearUtils.revokeUrl, {
+      method: "POST",
+      headers,
+    });
+  }
+
+  static async getInstalledWorkspace(accessToken: string) {
+    const client = new LinearClient({ accessToken });
+    return client.organization;
+  }
+
+  /**
+   *
+   * @param url Linear resource url
+   * @param actor User attempting to unfurl resource url
+   * @returns An object containing resource details e.g, a Linear issue details
+   */
+  static unfurl: UnfurlSignature = async (url: string, actor: User) => {
+    const resource = Linear.parseUrl(url);
+
+    if (!resource) {
+      return;
+    }
+
+    const integration = (await Integration.scope("withAuthentication").findOne({
+      where: {
+        service: IntegrationService.Linear,
+        teamId: actor.teamId,
+        "settings.linear.workspace.name": resource.workspace,
+      },
+    })) as Integration<IntegrationType.Embed>;
+
+    if (!integration) {
+      return;
+    }
+
+    try {
+      const client = new LinearClient({
+        accessToken: integration.authentication.token,
+      });
+      const issue = await client.issue(resource.id);
+
+      if (!issue) {
+        return { error: "Resource not found" };
+      }
+
+      const [author, state, labels] = await Promise.all([
+        issue.creator,
+        issue.state,
+        issue.paginate(issue.labels, {}),
+      ]);
+
+      if (!author || !state || !labels) {
+        return { error: "Failed to fetch auxiliary data from Linear" };
+      }
+
+      return {
+        type: UnfurlResourceType.Issue,
+        url: issue.url,
+        id: issue.identifier,
+        title: issue.title,
+        description: issue.description ?? null,
+        author: {
+          name: author.name,
+          avatarUrl: author.avatarUrl ?? "",
+        },
+        labels: labels.map((label) => ({
+          name: label.name,
+          color: label.color,
+        })),
+        state: {
+          type: state.type,
+          name: state.name,
+          color: state.color,
+        },
+        createdAt: issue.createdAt.toISOString(),
+        transformed_unfurl: true,
+      } satisfies UnfurlIssueAndPR;
+    } catch (err) {
+      Logger.warn("Failed to fetch resource from Linear", err);
+      return { error: err.message || "Unknown error" };
+    }
+  };
+
+  /**
+   * Parses a given URL and returns resource identifiers for Linear specific URLs
+   *
+   * @param url URL to parse
+   * @returns {object} Containing resource identifiers - `workspace`, `type`, `id` and `name`.
+   */
+  private static parseUrl(url: string) {
+    const { hostname, pathname } = new URL(url);
+    if (hostname !== "linear.app") {
+      return;
+    }
+
+    const parts = pathname.split("/");
+    const workspace = parts[1];
```

Comment:
Similarly, here this is called `workspaceKey` and should compare to `workspace.key`